Algorithm1 wordfreq update

ProjectZuki opened the pull request:
## Additions
Added functionality to word frequency.quorum to initialize hash table of stop words on initialization of WordFrequency.
Optional function added to remove stopwords from WordFrequency.

## Changes
Function Calculate() has been changed to Calculate_from_file() because there will be another method to calculate from a text giving us both options.

ProjectZuki committed:
Updated WordFrequency.quorum for stopwords

Added constructor that on create initializes a hashtable of stopwords (default name "stopwords-en.txt").

WordFrequency will only account for words that are not stopwords

ProjectZuki committed:
Update .gitignore

Dorffasaurus committed:
Added CalculateFromArray

ProjectZuki committed:
Added main.quorum

main.quorum added to test functionality of wordfrequency.quorum. wizard_of_oz.txt file added, where all punctuation except single quotes (e.g. words like don't, it's, i'm, etc...) are removed.

ProjectZuki committed:
Added main.quorum

This time i _actually_ added Main.quorum because the computer (definitely not human error) did not do so

Dorffasaurus commented on WordCountAlgo/wizard_of_oz.txt:
Please remove this

ProjectZuki replied:
removed wizard_of_oz.txt

ProjectZuki committed:
Delete wizard_of_oz.txt

Dorffasaurus commented on WordCountAlgo/Main.quorum:
remove the main.quorum file

ProjectZuki replied:
removed main.quorum

ProjectZuki committed:
Removed main.quorum

Dorffasaurus commented on WordCountAlgo/WordFrequency.quorum:
```diff
@@ -12,12 +12,52 @@ use Libraries.Language.Errors.UndefinedObjectError
+    private HashTable<text, boolean> StopWords
+
+    // constructor to initialize the list of stopwords
+    on create
+        // open stopwords file for reading
+        File stopwordsfile
+        FileReader stopwordsreader
+        stopwordsfile:SetPath("clean_stopwords-en.txt")
+        stopwordsreader:OpenForRead(stopwordsfile)
```
instead of requiring a file on create can we use a getter and setter function to set the stopwords.

Dorffasaurus commented on WordCountAlgo/WordFrequency.quorum:
```diff
@@ -44,19 +84,55 @@ class WordFrequency
+        end
+    end
+
 
     public action GetTotalWords returns integer
         Iterator<text> wordMapItr = WordCountMap:GetKeyIterator()
```
can we add a class variable that tracks how many unique words have been added? Then we can get rid of this function and just add a getter to return it.

ProjectZuki committed:
Updated WordFrequency.quorum

Modified SetFrequencyMap() to check if input text is filepath, if not will process as text (e.g. tweet, article)

Removed RemoveStopWords(), as SetFrequencyMap() will automatically remove them

Removed constructor, for same reason.

SetStopWords() requires filepath (text) for stopwords file

Modified GetTotalWords -> GetWordCount() to return total words.

ProjectZuki commented:
Updated the contents of WordFrequency.quorum

### Changes
Modified SetFrequencyMap() to check if input text is filepath, if not will process as text (e.g. tweet, article)

Removed RemoveStopWords(), as SetFrequencyMap() will automatically remove them

Removed constructor, for same reason.

SetStopWords() requires filepath (text) for stopwords file

Modified GetTotalWords -> GetWordCount() to return total words.

Dorffasaurus reviewed (approved):
Thanks for your work, appreciate the changes.
